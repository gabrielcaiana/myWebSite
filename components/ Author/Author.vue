<template>
  <div class="w-full px-2 xs:mb-6 md:mb-12 article-card">
    <NuxtLink
      :to="`/articles/author/${author.name}`"
      class="flex transition-shadow duration-150 ease-in-out shadow-sm hover:shadow-md xxlmax:flex-col"
    >
      <img 
        v-if="author.image"
        class="h-48 xxlmin:w-1/2 xxlmax:w-full object-cover"
        :src="author.image"
        :alt="author.name"
      />
      <div class="flex flex-col m-4">
        <h4 class="font-semibold">Author</h4>

        <p class="author-name">{{ author.name }}</p>
        <p class="author-bio">{{ author.bio }}</p>
      </div>
    </NuxtLink>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  name: 'Author',
  props: {
    author: {
      type: Object,
      required: true,
    },
  },
})
</script>
