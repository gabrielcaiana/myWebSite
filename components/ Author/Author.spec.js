import { mount, RouterLinkStub } from "@vue/test-utils"
import Author from "./Author.vue"

describe("Author", () => {

  let wrapper

  beforeAll(() => {
    wrapper = mount(Author, {
      stubs: {
        NuxtLink: RouterLinkStub
      },
      propsData: {
        author: {
          name: "Gabriel Caiana",
          bio: "I`am front-end developer",
          image: "images/me.jpg"
        }
      }
    })
  })

  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it("create Snapshot", () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it("should render the text", () => {
    expect(wrapper.find(".author-name").text()).toBe("Gabriel Caiana")
    expect(wrapper.find(".author-bio").text()).toBe("I`am front-end developer")
  })  
})
