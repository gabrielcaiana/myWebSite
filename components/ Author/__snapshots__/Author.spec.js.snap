// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author create Snapshot 1`] = `
<div
  class="w-full px-2 xs:mb-6 md:mb-12 article-card"
>
  <a
    class="flex transition-shadow duration-150 ease-in-out shadow-sm hover:shadow-md xxlmax:flex-col"
  >
    <img
      alt="Gabriel Caiana"
      class="h-48 xxlmin:w-1/2 xxlmax:w-full object-cover"
      src="images/me.jpg"
    />
     
    <div
      class="flex flex-col m-4"
    >
      <h4
        class="font-semibold"
      >
        Author
      </h4>
       
      <p
        class="author-name"
      >
        Gabriel Caiana
      </p>
       
      <p
        class="author-bio"
      >
        I\`am front-end developer
      </p>
    </div>
  </a>
</div>
`;
